File: server/src/main/java/com/edutech/progressive/entity/Warehouse.java
package com.edutech.progressive.entity;

import javax.persistence.*;

@Entity
public class Warehouse implements Comparable<Warehouse> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int warehouseId;
    
    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "supplierId")
    private Supplier supplier;
    private String warehouseName;
    private String location;
    private int capacity;

    public Warehouse() {
    }

    public Warehouse(int warehouseId, int supplierId, String warehouseName, String location, int capacity) {
        this.warehouseId = warehouseId;
        this.supplier.setSupplierId(supplierId);
        this.warehouseName = warehouseName;
        this.location = location;
        this.capacity = capacity;
    }

    public int getWarehouseId() {
        return warehouseId;
    }

    public void setWarehouseId(int warehouseId) {
        this.warehouseId = warehouseId;
    }

    public Supplier getSupplier() {
        return supplier;
    }

    public void setSupplier(Supplier supplier) {
        this.supplier = supplier;
    }

    public String getWarehouseName() {
        return warehouseName;
    }

    public void setWarehouseName(String warehouseName) {
        this.warehouseName = warehouseName;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }

    @Override
    public int compareTo(Warehouse otherWarehouse) {
        // Implement comparison logic based on warehouse capacity
        return Double.compare(otherWarehouse.getCapacity(), this.getCapacity());
    }
}

File: client/src/app/supplylink/types/Warehouse.ts
import { Supplier } from "./Supplier";
export class Warehouse {
    warehouseId: number;
    supplierId: string;
    warehouseName: string;
    location: string;
    capacity: number;
    constructor(warehouseId: any,supplierId: any,warehouseName: any,location: any,capacity: any){
        this.warehouseId = warehouseId;
        this.supplierId = supplierId;
        this.warehouseName = warehouseName;
        this.location = location;
        this.capacity = capacity;
    }

    displayInfo():void{
        console.log("Warehouse ID : " + this.warehouseId);
        console.log("Supplier ID : " + this.supplierId);
        console.log("Warehouse Name : " + this.warehouseName);
        console.log("Location : " + this.location);
        console.log("Capacity : " + this.capacity);
        
    }

 
}